cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -446,11 +446,18 @@ def get_user_specific_instrument_list(app):
                 email = tokenHelper.get_token_user_email_address(decoded_token)
 
         out_instrument_list = []
-        for instrument_factory in importer.instrument_factory_list:
-            instrument = instrument_factory()
+        for instrument_factory in importer.instrument_factory_iter:
+            if hasattr(instrument_factory, 'instrument_query'):
```

Comment:
What makes this condition necessary?

Reply:
Are there different factories?

Reply:
> Are there different factories?

The factories in "old" plugins don't have these attributes attached